File: leetcodepractice/151-reverse-words-in-a-string/reverse-words-in-a-string.js
/**
 * @param {string} s
 * @return {string}
 */
var reverseWords = function(s) {
    const sArray = s.split(' ');
    const newArray = [];

    for (let i = s.length - 1; i >= 0; i--){
        if (sArray[i]) newArray.push(sArray[i]);
    }

    return newArray.join(' ');
};

File: 151-reverse-words-in-a-string/reverse-words-in-a-string.py
class Solution:
    def reverseWords(self, s: str) -> str:
        reverse_order = []
        split = s.split(' ')
        
        for i in range(len(split) - 1, -1 ,-1):
            if split[i] != "":
                reverse_order.append(split[i])
        
        return " ".join(reverse_order)